tools/docker/start.sh

```diff
@@ -12,23 +12,28 @@
 set -ex
 
 REMOTE=${1:-https://github.com/web-platform-tests/wpt}
-BRANCH=${2:-master}
-REV=${3:-FETCH_HEAD}
+REF=${2:-master}
+REVISION=${3:-FETCH_HEAD}
 BROWSER=${4:-all}
 
 cd ~
 
-# Initially we just fetch 50 commits in order to save several minutes of fetching
-git clone ${REMOTE} --single-branch --branch ${BRANCH} --no-checkout -q --depth=50 web-platform-tests
+mkdir web-platform-tests
 cd web-platform-tests
-if [[ ! `git rev-parse --verify -q ${REV}` ]];
+git init
+git remote add origin ${REMOTE}
+
+# Initially we just fetch 50 commits in order to save several minutes of fetching
+git fetch --quiet --depth=50 origin ${REF}
```

Comment:
Technically we could pass in the base ref or the number of commits in the branch to ensure that the full PR is pulled in the unlikley event there are > 50 commits.

Reply:
The initial fetch is a provision for tasks run against pushes to `master`. In those cases, we don't have another base reference or a specific history length to work with. The good news is that this isn't a concern for pull requests [now that this patch uses GitHub's `ref/pull/XXX/merge` reference](https://github.com/web-platform-tests/wpt/pull/12657#issuecomment-417786800).  Thanks to the merge commit, we only need to fetch at a depth of 2 to generate a complete diff, regardless of the number of commits in the pull request.

Reply:
You're right, but this is a little different to what we do on Travis. The difference isn't very important and doesn't block this PR but does impose some limitations if we ever want to use the contents of the actual PR commits in the future.

For travis we use the GitHub-provided branch base to get the whole list of commits on the branch, and use that to figure out the tests that changed. Here we are falling back on the fact that the merge commit isn't on the master branch and the diff between that and master will produce the same list of tests. Which is fine, as long as we don't actually want the PR commits themselves for any reason.